corehq/apps/translations/integrations/transifex/client.py

```diff
@@ -60,19 +60,46 @@ def _create_with_form(cls, content, resource_id, language_id=None):
             raise UploadException(upload.errors[0]["detail"], upload.errors)
         return upload
 
-    def update_resource_slug(self, old_resource_slug, new_resource_slug):
-        url = "https://www.transifex.com/api/2/project/{}/resource/{}".format(
-            self.project, old_resource_slug)
-        data = {'slug': new_resource_slug}
-        headers = {'content-type': 'application/json'}
-        return requests.put(
-            url, data=json.dumps(data), auth=self._auth, headers=headers,
+    def move_resource(self, old_resource_slug, new_resource_slug):
```

Comment:
I misunderstood somewhat the original action of this function. This is used only by `ProjectMigrator`, which already takes steps on its own to compile and upload source strings and translations for the new resource. As such, this can simply:
1. get the old resource
2. create the new resource (using the old resource name)
3. delete the old resource

Reply:
Hey @nospame 

if memory serves, I added this function so it can be used to run migrations manually as shared by the [commit message](https://github.com/dimagi/commcare-hq/commit/4001c33d4e3e4bbd283563ab646319b8cd620ba8)
I don't see the use of it in `ProjectMigrator`.

Feel free to remove it if its a blocker or add a comment in the function that you think would be helpful for future readers.

Reply:
I believe by re-creating the resource and deleting the old one, you lose the history of action on the resource on transifex.
So, I am not sure if changing how this works is the right thing to do here.

Reply:
The main problem is that slug is now an immutable field on a resource, so it cannot be changed as before.

As for functionality, I was looking in `_update_menus_and_forms_sheet` which itself calls `_upload_new_translations`. https://github.com/dimagi/commcare-hq/blob/248d6e80368fff521da14c04923bc487e8fe5211/corehq/apps/translations/integrations/transifex/project_migrator.py#L68
However I think it will cause issues to delete the resource as part of `move_resource`, since `_update_menus_and_forms_sheet` also wants to use `client.get_translation` which would fail if the resource has been deleted. So I think the single call to `client.move_resource` will actually change into something like `client.copy_resource`, which can be followed by `client.delete_resource` after all the translations have been updated and moved. Does that seem right?

Reply:
The name `_upload_new_translations` is probably misleading. It seems it is only updating the file `Menus_and_forms` and not all files.

Reply:
That makes sense. I'll look into it further and try to find a good solution. I think the _right answer_ might ultimately mean having to rewrite a chunk of the transifex project migrator code, which is a little outside my current scope. But possibly something like the initial change, as involved as that is (creating new resource, getting old source strings and translations, uploading new source strings and translations, deleting the old resource) would work as a stop-gap.

Reply:
Based on your explanation offline this seems like it could be a useful feature, but not easy to re-implement with the new API. @Ben-Talbot and I discussed yesterday and based on that decided to hide the functionality without fully removing it, for now. I expect it may simply be removed in the future if we don't end up using it. If you (or anyone else) has strong feelings otherwise, I'd be happy to hear them. https://github.com/dimagi/commcare-hq/pull/34242/commits/14809f0520c8dcd53d96a5a8c4eb03601e8c3a20